src/core/namespace.c

```diff
@@ -287,10 +288,15 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
+                r = lstat(m->path, &inaccessible_target);
```

Comment:
But to evaluate if it is a symlink, we would need to check every component of the path...

The path `$CHROOT/lib64/mozilla/plugins/` in the chroot might be wrongly resolved to `/usr/lib64/mozilla/plugins/` if `$CHROOT/lib64` is a symlink to `/usr/lib64`. Ideally, it should be a relative symlink, but images exist where it is an absolute symlink. So I would like if we support that...